### PR TITLE
Feat: issues filter

### PR DESCRIPTION
**Pull request by MarceloRobert**
- Adds a clickable filter for issues card
- Changes how unknown issues are counted for boots and tests
- Changes the issue url to be shown as an icon to the left of the issue listed in the card instead of being a clickable in the text itself
- Fixes build issues card listing all issues (not only build issues)

## How to test:
Go to a treeDetails page that has issues such as this [localhost example](http://localhost:5173/tree/3ed48e91c838a2d00d06c208590a48ce96ce0d00?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentPageTab=global.builds&diffFilter=%7B%7D&treeInfo=%7B%22gitBranch%22%3A%22linux-4.4.y-cip%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Flinux%2Fkernel%2Fgit%2Fcip%2Flinux-cip.git%22%2C%22treeName%22%3A%22cip%22%2C%22commitName%22%3A%22v4.4.302-cip72-2386-g3ed48e91c838%22%2C%22headCommitHash%22%3A%223ed48e91c838a2d00d06c208590a48ce96ce0d00%22%7D&intervalInDays=7) and compare it to the [equivalent on staging](https://staging.dashboard.kernelci.org:9000/tree/3ed48e91c838a2d00d06c208590a48ce96ce0d00?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=maestro&currentPageTab=global.builds&diffFilter=%7B%7D&treeInfo=%7B%22gitBranch%22%3A%22linux-4.4.y-cip%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Flinux%2Fkernel%2Fgit%2Fcip%2Flinux-cip.git%22%2C%22treeName%22%3A%22cip%22%2C%22commitName%22%3A%22v4.4.302-cip72-2386-g3ed48e91c838%22%2C%22headCommitHash%22%3A%223ed48e91c838a2d00d06c208590a48ce96ce0d00%22%7D&intervalInDays=7).

### CODE REVIEW

File: backend/kernelCI_app/views/treeCommitsHistory.py [hunk 1]
```diff
@@ -54,8 +54,9 @@ def __treat_unknown_filter(self, table_field, op, value_name, filter):
         clause = table_field
         is_null_clause = f"{table_field} IS NULL"
         has_null_value = False
-        if (isinstance(filter['value'], str)):
-            filter['value'] = '' if filter['value'] in ['NULL', 'Unknown'] else filter['value']
+
+        if (isinstance(filter['value'], str) and filter['value'] in ['NULL', 'Unknown']):
```
Comment (Francisco2002) [inline backend/kernelCI_app/views/treeCommitsHistory.py:58]:
```suggestion
        if (filter['value'] is str and filter['value'] in ['NULL', 'Unknown']):
```

File: dashboard/src/components/Cards/IssuesList.tsx [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
 import ListingItem, { ItemType } from '@/components/ListingItem/ListingItem';
 
 import ColoredCircle from '@/components/ColoredCircle/ColoredCircle';
 import { NoIssueFound } from '@/components/Issue/IssueSection';
-import type { TIssue } from '@/types/general';
+import type { TFilter, TFilterObjectsKeys, TIssue } from '@/types/general';
+
+import FilterLink from '@/components/Tabs/FilterLink';
+
+import LinkWithIcon from '../LinkWithIcon/LinkWithIcon';
```
Comment (Francisco2002) [inline dashboard/src/components/Cards/IssuesList.tsx:18]: relative import

File: backend/kernelCI_app/views/treeCommitsHistory.py (hunk 1 above)
Comment (MarceloRobert) [reply backend/kernelCI_app/views/treeCommitsHistory.py]: These are actually not equivalent, `is str` would compare if `filter['value']` is literally `str`, while `isinstance` would check if the class of `filter['value']` is str

**Review by murilx — APPROVED** [review]

**Review by WilsonNet — APPROVED** [review]
looks good it is working fine

**Pull request merged**